tests/cpu/pmp_stack_guard/main.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2016 Loci Controls Inc.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief Test application for the pmp_stack_guard pseudo-module
+ *
+ * @author Ian Martin <ian@locicontrols.com>
+ * @author Bennet Blischke <bennet.blischke@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "cpu.h"
+#include "thread.h"
+#include "thread_arch.h"
+#include "pmp.h"
+
+#define CANARY_VALUE 0xdeadbeef
+
+static struct {
+    char overflow_buffer[128];
+    unsigned int canary;
+    char stack[THREAD_STACKSIZE_MAIN];
+} buf;
+
+
+static inline unsigned int __get_SP(void) {
```

Comment:
I find myself quite often needing such little utility functions, maybe RIOT should collect them to a common interface / module.